packages/react-components/src/components/table-output-component.tsx

```diff
@@ -508,13 +516,59 @@ export class TableOutputComponent extends AbstractOutputComponent<TableOutputPro
         return lines[0].index;
     }
 
-    private async findNextMatchedEvent() {
+    private async findMatchedEvent(direction: Direction) {
         let isFound = false;
         if (this.gridApi) {
             this.gridApi.deselectAll();
+            let rowNodes: RowNode[] = [];
+            const blockSize = this.props.cacheBlockSize ? this.props.cacheBlockSize : 200;
+
+            // make sure that both index are either both -1 or both have a valid number
+            if (this.selectStartIndex !== -1 && this.selectEndIndex === -1) {
+                this.selectEndIndex = this.selectStartIndex;
+            }
+            if (this.selectEndIndex !== -1 && this.selectStartIndex === -1) {
+                this.selectStartIndex = this.selectStartIndex;
+            }
+
+            /*
+             * current block is where the selection is either
+             * - the block that contains the selection start or
+             * - current visible block
+             */
+            let currentBlock = Math.floor(this.gridApi?.getLastDisplayedRow() / blockSize);
+            let currRowIndex = this.gridApi?.getLastDisplayedRow() + 1;
+            if (this.selectStartIndex !== -1) {
+                currentBlock = Math.floor(this.selectStartIndex / blockSize);
+                currRowIndex = Math.max(this.selectStartIndex, this.selectEndIndex);
+                if (direction === Direction.PREVIOUS) {
+                    Math.min(this.selectStartIndex, this.selectEndIndex);
+                }
+            }
+
             this.gridApi.forEachNode(rowNode => {
-                if (rowNode.rowIndex > Math.max(this.selectStartIndex, this.selectEndIndex) && rowNode.data
-                    && rowNode.data['isMatched'] && !isFound) {
+                // make sure to search only in current block and not in full cache
+                if (currentBlock === Math.floor(rowNode.rowIndex / blockSize)) {
```

Comment:
This prevents considering contiguous blocks. I propose to use a scheme that only considers contiguous row index regardless of their block.